bigbluebutton-html5/imports/api/captions/server/handlers/captionReply.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'underscore';
+import Logger from '/imports/startup/server/logger';
+import { check } from 'meteor/check';
+import { inReplyToHTML5Client } from '/imports/api/common/server/helpers';
+import addCaption from '../modifiers/addCaption';
+
+export default function handleCaptionReply({ payload }) {
+  if (!inReplyToHTML5Client({ payload })) {
+    return;
+  }
+
+  const SERVER_CONFIG = Meteor.settings.app;
+  const CAPTION_CHUNK_LENGTH = SERVER_CONFIG.captionsChunkLength || 1000;
+
+  const meetingId = payload.meeting_id;
+  const locale = payload.locale;
+  const captionHistory = payload.caption_history;
+
+  check(meetingId, String);
+  check(captionHistory, Object);
+
+  let captionsAdded = [];
+  _.each(captionHistory, (caption, locale) => {
+    let ownerId = caption[0];
+    let captions = caption[1].slice(0);
+    let chunks = [];
+
+    while (captions.length > 0) {
+      if (captions.length > CAPTION_CHUNK_LENGTH) {
+        chunks.push(captions.slice(0, CAPTION_CHUNK_LENGTH));
+        captions = captions.slice(CAPTION_CHUNK_LENGTH);
+      } else {
+        chunks.push(captions);
+        captions = captions.slice(captions.length);
+      }
+    }
+
```

Comment:
while was wrapped into if - else block here.
If the length of the chunk is 0 - we still need to push an empty object to chunks, otherwise it breaks the order.
https://github.com/bigbluebutton/bigbluebutton/blob/master/bigbluebutton-html5/imports/api/captions/server/modifiers/addCaptionsToCollection.js#L18-L20